lib/core/versions/0.9.0/RequestHandler.ts

```diff
@@ -190,22 +193,24 @@ export default class RequestHandler implements IRequestHandler {
 
   /**
    * Resolves the given long-form DID by resolving using operations found over the network first;
-   * if no operations found, the given create operation will is used to construct the DID state.
+   * if no operations found, the given create operation will be used to construct the DID state.
+   * 
+   * @returns [DID state, published]
    */
-  private async resolveLongFormDid (did: Did): Promise<DidState | undefined> {
+  private async resolveLongFormDid (did: Did): Promise<[DidState | undefined, boolean]> {
```

Comment:
So can't this boolean not be part of the DidState? Sorry I couldn't be part of the design discussion so feel free to continue if you want and fill me in later.

Reply:
The discussion was precise on this. TL;DR is that we could include `published` in `DidState` but it isn't necessarily cleaner, because it would have to "undefined" until higher layer finds out if it is published or not.

Reply:
Further discussion is with @mudiali, he raised a good point that we can potentially special case input to apply create operation if we wanted to `published` info in the `DidState`. Bottomline is that this current approach isn't necessarily bad because the `published` info isn't needed by operation processor. Since the other change will be much more involved, we decided keep the current approach. But we can follow the suggested pattern when we have new input required by the operation apply methods in the future.